File: python/course-repo/solution/02-idempotency-and-crash-safety/src/webhooks/worker.py
import asyncio
from concurrent.futures import ThreadPoolExecutor

from temporalio.client import Client
from temporalio.worker import Worker

from .activities import deliver_webhook
from .shared import TASK_QUEUE


async def main() -> None:
    client = await Client.connect("localhost:7233")
    with ThreadPoolExecutor(5) as executor:
        worker = Worker(
            client,
            task_queue=TASK_QUEUE,
            activities=[deliver_webhook],
            activity_executor=executor,
        )
        print(f"Worker running on task queue '{TASK_QUEUE}'")
        await worker.run()


if __name__ == "__main__":
    asyncio.run(main())

File: typescript/course-repo/solution/05-heartbeats-and-checkpointing/src/worker.ts
import { Worker, NativeConnection } from '@temporalio/worker';
import * as activities from './activities';
import { TASK_QUEUE } from './shared';

async function main() {
  const connection = await NativeConnection.connect({ address: 'localhost:7233' });
  const worker = await Worker.create({
    connection,
    taskQueue: TASK_QUEUE,
    activities,
  });
  console.log(`Worker running on task queue '${TASK_QUEUE}'`);
  await worker.run();
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
